models/public/wavernn/wavernn_to_onnx.py

```diff
@@ -0,0 +1,264 @@
+    # Instantiate WaveRNN Model
+    voc_model = WaveRNN(rnn_dims=hp.voc_rnn_dims,
+                        fc_dims=hp.voc_fc_dims,
+                        bits=hp.bits,
+                        pad=hp.voc_pad,
+                        upsample_factors=hp.voc_upsample_factors,
+                        feat_dims=hp.num_mels,
+                        compute_dims=hp.voc_compute_dims,
+                        res_out_dims=hp.voc_res_out_dims,
+                        res_blocks=hp.voc_res_blocks,
+                        hop_length=hp.hop_length,
+                        sample_rate=hp.sample_rate,
+                        mode=hp.voc_mode).to(device)
+
+    voc_load_path = args.voc_weights
+    voc_model.load(voc_load_path)
+
+    voc_upsampler = WaveRNNUpsamplerONNX(voc_model, args.batched, hp.voc_target, hp.voc_overlap)
+    voc_infer = WaveRNNONNX(voc_model)
+
+    voc_model.eval()
+    voc_upsampler.eval()
+    voc_infer.eval()
+
+
+    opset_version = 10
+
+    with torch.no_grad():
+        mels = np.load(args.mel)
+
+        mel = (mels + 4) / 8
+        np.clip(mel, 0, 1, out=mel)
+        wav = reconstruct_waveform(mels, n_iter=60)
+
+        save_wav(wav, "./onnx/wav_from_mel_griffin_lim.wav")
+
+        '''
+        FIRST STEP: Upsample mels to the output wave shape
+        '''
+        mels = torch.from_numpy(mels)
+        mels = mels.unsqueeze(0)
+
+        wave_len = (mels.size(-1) - 1) * voc_model.hop_length
+
+        mels = voc_upsampler.pad_tensor(mels)
+
+        mels_onnx = mels.clone()
+
+        print(mels_onnx.shape)
+        torch.onnx.export(voc_upsampler, mels_onnx, "./onnx/wavernn_upsampler.onnx",
+                          opset_version=opset_version,
+                          do_constant_folding=True,
+                          input_names=["mels"],
+                          output_names=["upsample_mels", "aux"])
+
+
+        mels, aux = voc_upsampler(mels)
+        mels = mels[:, 550:-550, :]
+
+        mels, aux = voc_upsampler.fold(mels, aux)
+
+        print("Mels/Aux fold shape: {0}/{1}".format(mels.shape, aux.shape))
+
+        output = []
+
+        h1, h2, x = voc_infer.get_initial_parameters(mels)
+
+        aux_split = voc_infer.split_aux(aux)
+
+        b_size, seq_len, _ = mels.size()
+
+        '''
+        SECOND STEP: Iterativelly generate wavs
+        '''
+
+        for i in range(seq_len):
+            m_t = mels[:, i, :]
+
+            a1_t, a2_t, a3_t, a4_t = \
+                (a[:, i, :] for a in aux_split)
+
+            if i == 0:
+                rnn_input = (m_t, a1_t, a2_t, a3_t, a4_t, h1, h2, x)
+                torch.onnx.export(voc_infer, rnn_input, "./onnx/wavernn_rnn.onnx",
+                                  opset_version=opset_version,
+                                  do_constant_folding=True,
+                                  input_names=["m_t", "a1_t", "a2_t", "a3_t", "a4_t", "h1", "h2", "x"],
+                                  output_names=["logits", "h1", "h2"])
```

Comment:
This doesn't quite make sense. If you only export the ONNX file on iteration 0, then why is this a loop at all? Just export it and be done with it.